src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/task/createAndRunTaskTool.ts

```diff
@@ -84,18 +84,25 @@ export class CreateAndRunTaskTool implements IToolImpl {
 		if (!exists) {
 			await this._fileService.createFile(tasksJsonUri, VSBuffer.fromString(tasksJsonContent), { overwrite: true });
 			_progress.report({ message: 'Created tasks.json file' });
-			await timeout(200);
 		} else {
 			// add to the existing tasks.json file
 			const content = await this._fileService.readFile(tasksJsonUri);
 			const tasksJson = JSON.parse(content.value.toString());
 			tasksJson.tasks.push(newTask);
 			await this._fileService.writeFile(tasksJsonUri, VSBuffer.fromString(JSON.stringify(tasksJson, null, '\t')));
 			_progress.report({ message: 'Updated tasks.json file' });
-			await timeout(200);
 		}
 		_progress.report({ message: new MarkdownString(localize('copilotChat.fetchingTask', 'Resolving the task')) });
-		const task = (await this._tasksService.tasks())?.find(t => t._label === args.task.label);
+
+		let task: Task | undefined;
+		const start = Date.now();
+		while (Date.now() - start < 5000 && !token.isCancellationRequested) {
+			task = (await this._tasksService.tasks())?.find(t => t._label === args.task.label);
+			if (task) {
+				break;
+			}
+			await timeout(100);
+		}
```

Comment:
If there's an event for adding tasks that would be a better approach, we can then do:

```ts
const found = await raceTimeout(new Promise<boolean>(r => {
  // Register and listen to event, resolve only if it matches
}, 5000);
// Handle cancel
```